crates/bws/src/command/mod.rs

```diff
@@ -0,0 +1,72 @@
+pub(crate) mod project;
+pub(crate) mod secret;
+
+use std::{path::PathBuf, str::FromStr};
+
+use bitwarden::auth::AccessToken;
+use bitwarden_cli::Color;
+use clap::CommandFactory;
+use clap_complete::Shell;
+use color_eyre::eyre::{bail, Result};
+
+use crate::{config, Cli, Output, ProfileKey};
+
+pub(crate) struct OutputSettings {
+    pub(crate) output: Output,
+    pub(crate) color: Color,
+}
+
+impl OutputSettings {
+    pub(crate) fn new(output: Output, color: Color) -> Self {
+        OutputSettings { output, color }
+    }
+}
+
+pub(crate) fn completions(shell: Option<Shell>) -> Result<()> {
+    let Some(shell) = shell.or_else(Shell::from_env) else {
+        eprintln!("Couldn't autodetect a valid shell. Run `bws completions --help` for more info.");
+        std::process::exit(1);
+    };
```

Comment:
I think we should try and avoid having `std::process::exit` outside the main file. One potential refactor here would be to have this function return an Error and handle the error in `main.rs`.

Reply:
Updated in https://github.com/bitwarden/sdk/pull/809/commits/f599033914f1e8740cac3cd64f6f1c0927915500.